src/main/java/net/openhft/chronicle/hash/ChronicleHashBuilder.java

```diff
@@ -750,4 +750,11 @@ H recoverPersistedTo(
      */
     @Deprecated
     Object privateAPI();
+
+    /**
+     * A {@link ChronicleHash} created using this builder is closed using a JVM shutdown hook by
+     * default. This method lets you avoid the close from shutdown hook and explicitly control
+     * when the {@link ChronicleHash} is closed.
+     */
+    void skipCloseByShutdownHook();
```

Comment:
I'm concerned about the safety of this solution. Maybe instead provide an ability to inject some action before the main ChronicleMap's shutdown hook?

Reply:
Done